### PR TITLE
drivers/clock_control: stm32u5: Use LL utility to set MSI as PLL srce

### PR DESCRIPTION
**Pull request by erwango**
Recent changes in this function made the code now compatible
with the use of LL API to perform this configuration.
Use it to simplify the driver.

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

### CODE REVIEW

File: drivers/clock_control/clock_stm32_ll_u5.c [hunk 1]
```diff
@@ -303,82 +303,19 @@ void config_src_sysclk_pll(LL_UTILS_ClkInitTypeDef s_ClkInitStruct)
...
-
-	LL_RCC_SetAHBPrescaler(s_ClkInitStruct.AHBCLKDivider);
-	LL_RCC_SetAPB1Prescaler(s_ClkInitStruct.APB1CLKDivider);
-	LL_RCC_SetAPB2Prescaler(s_ClkInitStruct.APB2CLKDivider);
-	LL_RCC_SetAPB3Prescaler(s_ClkInitStruct.APB3CLKDivider);
+#if STM32_PLL_SRC_MSIS
+	set_up_clk_msis();
 
-	LL_SetSystemCoreClock(CONFIG_SYS_CLOCK_HW_CYCLES_PER_SEC);
+	/* Switch to PLL with HSI as clock source */
```
Comment (FRASTM) [inline drivers/clock_control/clock_stm32_ll_u5.c:311]: could it be `/* Switch to PLL with MSI as clock source */`

**Review by FRASTM — APPROVED** [review]

File: drivers/clock_control/clock_stm32_ll_u5.c (hunk 1 above)
Comment (erwango) [reply drivers/clock_control/clock_stm32_ll_u5.c]: Fixed, txs

File: drivers/clock_control/clock_stm32_ll_u5.c [hunk 2]
```diff
@@ -303,82 +303,19 @@ void config_src_sysclk_pll(LL_UTILS_ClkInitTypeDef s_ClkInitStruct)
...
-
-	LL_RCC_SetAHBPrescaler(s_ClkInitStruct.AHBCLKDivider);
-	LL_RCC_SetAPB1Prescaler(s_ClkInitStruct.APB1CLKDivider);
-	LL_RCC_SetAPB2Prescaler(s_ClkInitStruct.APB2CLKDivider);
-	LL_RCC_SetAPB3Prescaler(s_ClkInitStruct.APB3CLKDivider);
+#if STM32_PLL_SRC_MSIS
+	set_up_clk_msis();
 
-	LL_SetSystemCoreClock(CONFIG_SYS_CLOCK_HW_CYCLES_PER_SEC);
+	/* Switch to PLL with MSI as clock source */
```
Comment (ABOSTM) [inline drivers/clock_control/clock_stm32_ll_u5.c:311]: It looks to me like we loose VCO input configuration and EPOD configuration. Is it ?

**Review by ABOSTM — CHANGES_REQUESTED** [review]

File: drivers/clock_control/clock_stm32_ll_u5.c (hunk 2 above)
Comment (erwango) [reply drivers/clock_control/clock_stm32_ll_u5.c:311]: Indeed, maybe I was a bit fast in removing all this. This being said, it works in all config that I have been able to test.
One thing is that none of these is specific to MSI, so it should actually be extented to other PLL sources.
I'll have a look

Comment (erwango) [reply drivers/clock_control/clock_stm32_ll_u5.c:311]: @ABOSTM Looks like implementing this for all PLL will be easier on the curated U5 driver version proposed in https://github.com/zephyrproject-rtos/zephyr/pull/41585. So I propose to log this in an issue and implemented on top of it.

**Review by ABOSTM — APPROVED** [review]

**Pull request merged**